src/components/connection_handler/test/connection_handler_impl_test.cc

```diff
@@ -412,13 +414,13 @@ TEST_F(ConnectionHandlerTest, OnMalformedMessageCallback) {
 
   InSequence seq;
   EXPECT_CALL(mock_connection_handler_observer,
-              OnServiceEndedCallback(connection_key_,kMobileNav, kMalformed)).Times(0);
+              OnServiceEndedCallback(connection_key_,kRpc, kMalformed)).Times(1);
```

Comment:
It's strange that we close the RPC at the begging. On protocol layer that mean close the whole session with all other services.